Content/posts/auto-generating-auto-hosting-and-auto-updating-docc-documentation.md

```diff
@@ -0,0 +1,83 @@
+---
+date: 2022-06-03 12:00
+title: Auto-generating, Auto-hosting, and Auto-updating DocC Documentation
+description: We’ve been working on something that should help authors when publishing their package: the easiest way to host your package documentation.
+---
+
+Introduced at WWDC 2021, [DocC](https://developer.apple.com/documentation/docc) is Apple’s recommended way to provide package documentation.
+
+It’s easy to use, and the resulting documentation looks great. It generates documentation from documentation comments and article files written in Markdown that suit longer-form documentation. You can even use it to create beautiful interactive tutorials with images and step-by-step instructions. DocC generates either Xcode documentation archives or HTML and CSS files that you can host on a web server.
+
+Of course, once you have a directory full of HTML, that’s only half the battle. You then need to get it hosted and maybe even set up a CI task to handle that task so that your published documentation stays up-to-date as your development progresses.
+
+That’s where our latest feature will come in handy, and we’re launching it today!
+
+## Auto-generated, auto-hosted, and auto-updating
+
+As of today, our build system can now generate and host DocC documentation and make it available from your package’s page in the index.
+
+All we need is a little configuration data so that we know how best to build your docs, and you’ll find a new “Documentation” link in the sidebar.
+
+![A screenshot showing the documentation link on a package page.](/images/documentation-link.png)
+
+As a package author or maintainer, there are only two steps you need to take to have the Swift Package Index host your documentation.
+
+1. Add the [swift-docc-plugin](https://github.com/apple/swift-docc-plugin) documentation plugin to your `Package.swift` manifest file, if you haven’t done so already.
+2. Create a `.spi.yml` file and commit it in the root of your package repository telling our build system which targets to build documentation for.
+
+From there, we’ll take care of everything else. Every time you push new commits to your package, we’ll regenerate your documentation.
+
+### Adding the Swift DocC plugin
+
+You may already have taken this first step if you’ve worked with DocC locally. If not, then add the following lines to the end of your `Package.swift` manifest:
+
+```swift
+#if swift(>=5.6)
+  // Add the documentation compiler plugin if possible
+  package.dependencies.append(
+    .package(url: “https://github.com/apple/swift-docc-plugin“, from: “1.0.0“)
+  )
+#endif
+```
+
+Making the plugin dependency conditional means that nothing will break even if the `tools-version` of your manifest is for an earlier version of Swift than 5.6.
```

Comment:
We should probably mention that the package has to support Swift 5.6 in order for us to be able to generate docs.